src/ewatercycle/_forcings/caravan.py

```diff
@@ -123,15 +124,21 @@ def generate(
                 'YYYY-MM-DDTHH:MM:SSZ'.
             end_time: nd time of forcing in UTC and ISO format string e.g.
                 'YYYY-MM-DDTHH:MM:SSZ'.
-            basin_id: Caravan basin_id
             directory: Directory in which forcing should be written.
                 If not given will create timestamped directory.
             variables: Variables which are needed for model,
                 if not specified will default to all.
             shape: (Optional) Path to a shape file.
                 If none is specified, will be downloaded automatically.
-            **kwargs
+            dataset: Unused
+
+            **kwargs:
+                basin_id: str containing the wanted basin_id.
+
         """
+        assert type(dataset) == str
+        basin_id = kwargs["basin_id"]
```

Comment:
Can you add a check that gives the user a clear input error if the basin ID is missing?

Reply:
Once this is fixed and you've merged from main (failing test is fixed in there), I'll approve and merge the PR

Reply:
Ah forgot to remove line 139, the `basin_id` is now passed as a kwarg rather than using the dataset variable. 

Will merge from main and update changelog